### PR TITLE
ci: add gh workflow for unit test

### PR DESCRIPTION
**Pull request by aramase**

### CODE REVIEW

**Review by CecileRobertMichon — APPROVED** [review]
LGTM

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+permissions: read-all
+
+jobs:
+  test:
+    name: 'Unit Test'
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Set up Go 1.20
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
```
Comment (jackfrancis) [inline .github/workflows/pr.yml:16]: why shouldn't we use `actions/setup-go@v4.0.1` ?

Comment (aramase) [reply .github/workflows/pr.yml:16]: Best practice is to always pin to the sha digest because tags are mutable

Comment (jackfrancis) [reply .github/workflows/pr.yml:16]: if anyone out there is listening "don't use tags immutably!!!!!"

**Pull request merged**